test/SerialIOTest.cpp

```diff
@@ -3454,6 +3455,151 @@ TEST_CASE( "serial_iterator", "[serial][adios2]" )
+            unsigned dimensionality = i % 3 + 1;
+            unsigned len = i + 1;
+            Extent changingExtent( dimensionality, len );
+            E_y.resetDataset( { openPMD::Datatype::INT, changingExtent } );
+            std::vector< int > changingData(
+                std::pow( len, dimensionality ), dimensionality );
+            E_y.storeChunk(
+                changingData, Offset( dimensionality, 0 ), changingExtent );
+            iteration.close();
+        }
+    }
+
+    REQUIRE( auxiliary::directory_exists(
+        auxiliary::replace_last( file, "%V", "" ) ) );
+
+    {
+        Series readSeries(
+            file,
+            Access::READ_ONLY,
+            "{\"defer_iteration_parsing\": true}" );
+
+        size_t last_iteration_index = 0;
+        for( auto iteration : readSeries.readIterations() )
+        {
+            auto E_x = iteration.meshes[ "E" ][ "x" ];
+            REQUIRE( E_x.getDimensionality() == 1 );
+            REQUIRE( E_x.getExtent()[ 0 ] == extent );
+            auto chunk = E_x.loadChunk< int >( { 0 }, { extent } );
+            iteration.close();
+            for( size_t i = 0; i < extent; ++i )
+            {
+                REQUIRE( chunk.get()[ i ] == int( iteration.iterationIndex ) );
+            }
+
+            auto E_y = iteration.meshes[ "E" ][ "y" ];
+            unsigned dimensionality = iteration.iterationIndex % 3 + 1;
+            unsigned len = iteration.iterationIndex + 1;
+            Extent changingExtent( dimensionality, len );
+            REQUIRE( E_y.getExtent() == changingExtent );
+            last_iteration_index = iteration.iterationIndex;
+        }
+        REQUIRE( last_iteration_index == 9 );
+    }
+
+    {
+        Series readSeries(
+            file,
+            Access::READ_ONLY,
+            "{\"defer_iteration_parsing\": false}" );
+
+        size_t last_iteration_index = 0;
+        for( auto iteration : readSeries.readIterations() )
+        {
+            auto E_x = iteration.meshes[ "E" ][ "x" ];
+            REQUIRE( E_x.getDimensionality() == 1 );
+            REQUIRE( E_x.getExtent()[ 0 ] == extent );
+            auto chunk = E_x.loadChunk< int >( { 0 }, { extent } );
+            iteration.close();
+            for( size_t i = 0; i < extent; ++i )
+            {
+                REQUIRE( chunk.get()[ i ] == int( iteration.iterationIndex ) );
+            }
+
+            auto E_y = iteration.meshes[ "E" ][ "y" ];
+            unsigned dimensionality = iteration.iterationIndex % 3 + 1;
+            unsigned len = iteration.iterationIndex + 1;
+            Extent changingExtent( dimensionality, len );
+            REQUIRE( E_y.getExtent() == changingExtent );
+            last_iteration_index = iteration.iterationIndex;
+        }
+        REQUIRE( last_iteration_index == 9 );
+    }
+}
+
+TEST_CASE( "variableBasedSeries", "[serial][adios2]" )
+{
+    variableBasedSeries( "../samples/variableBasedSeries%V.bp" );
```

Comment:
Discussed offline: let's move this to JSON options

Reply:
I suggest moving this to a separate PR, since this will also be used for the other iteration layouts and there are further refinements to the JSON configs that I would like to do apart from this, including:
* pick backend via JSON
* caps-insensitivity
* clearly define and document precedences (environment variables > JSON options > automatic detection from filenames)
* extended patterns as in PIConGPU (that one maybe as a further separate PR)